packages/dcos-integration-test/extra/test_mesos.py

```diff
@@ -0,0 +1,135 @@
+    master_state_url = 'http://{}:{}/state'.format(cluster.masters[0], 5050)
+    r = requests.get(master_state_url)
+    logging.debug('Got %s with request for %s. Response: \n%s', r.status_code, master_state_url, r.text)
+    assert r.status_code == 200
+    state = r.json()
+
+    # Find the agent_id and container_id from master state
+    container_id = None
+    agent_id = None
+    for framework in state['frameworks']:
+        for task in framework['tasks']:
+            if app_id in task['id']:
+                container_id = task['statuses'][0]['container_status']['container_id']['value']
+                agent_id = task['slave_id']
+    if container_id is None:
+        raise Exception('Container ID not found for instance of app_id {}'.format(app_id))
+    if agent_id is None:
+        raise Exception('Agent ID not found for instance of app_id {}'.format(app_id))
+
+    # Find hostname and URL from agent_id
+    agent_hostname = None
+    for agent in state['slaves']:
+        if agent['id'] == agent_id:
+            agent_hostname = agent['hostname']
+    if agent_hostname is None:
+        raise Exception('Agent hostname not found for agent_id {}'.format(agent_id))
+    agent_v1_url = 'http://{}:{}/api/v1'.format(agent_hostname, 5051)
+    logging.debug('Located %s with containerID %s on agent %s', app_id, container_id, agent_hostname)
+
+    # Prepare nested container id data
+    nested_container_id = {
+        'value': 'debug-%s' % str(uuid.uuid4()),
+        'parent': {'value': '%s' % container_id}}
+
+    # Launch debug session and attach to output stream of debug container
+    output_headers = {
+        'Content-Type': 'application/json',
+        'Accept': 'application/json+recordio',
+        'Connection': 'keep-alive'
+    }
+    lncs_data = {
+        'type': 'LAUNCH_NESTED_CONTAINER_SESSION',
+        'launch_nested_container_session': {
+            'command': {'value': 'cat'},
+            'container_id': nested_container_id}}
+    launch_output = post(agent_v1_url, output_headers, json=lncs_data, stream=True)
+
+    # Attach to output stream of nested container
+    attach_out_data = {
+        'type': 'ATTACH_CONTAINER_OUTPUT',
+        'attach_container_output': {'container_id': nested_container_id}}
+    attached_output = post(agent_v1_url, output_headers, json=attach_out_data, stream=True)
+
+    # Attach to input stream of debug container and stream a message
+    input_headers = {
+        'Content-Type': 'application/json+recordio',
+        'Accept': 'application/json',
+        'Connection': 'keep-alive',
+        'Transfer-Encoding': 'chunked'
+    }
+    post(agent_v1_url, input_headers, data=input_streamer(nested_container_id))
+
+    # Verify the streamed output from the launch session
+    decoder = Decoder(lambda s: json.loads(s.decode("UTF-8")))
+    for chunk in launch_output.iter_content():
+        for r in decoder.decode(chunk):
+            if r['type'] == 'DATA':
```

Comment:
This test can pass if there are no records that are of the data type. Should we add a check so that this looper will fail if this if statement is never triggered?